crates/gitbutler-branch-actions/src/dependencies.rs

```diff
@@ -73,6 +67,41 @@ pub fn compute_workspace_dependencies(
     })
 }
 
+/// Get the commits that need to be processed for the stack dependencies caculation.
+///
+/// Commit IDs are in the order that they are applied (parent first).
+/// Merge commits to the target branch are not included.
+fn get_commits_to_process(
+    repo: &git2::Repository,
+    stack: &Stack,
+    target_sha: &git2::Oid,
+) -> Result<Vec<git2::Oid>, anyhow::Error> {
+    let commit_ids = repo
+        .l(stack.head(), LogUntil::Commit(*target_sha), true)
+        .context("failed to list commits")?
+        .into_iter()
+        .rev()
+        .filter_map(|commit_id| {
+            let commit = repo.find_commit(commit_id).ok()?;
+            if commit.parent_count() == 1 {
+                return Some(commit_id);
+            }
+
+            let has_integrated_parent = commit.parent_ids().any(|id| {
+                let (number_commits_ahead, _) = repo.graph_ahead_behind(id, *target_sha).unwrap();
+                number_commits_ahead == 0
```

Comment:
Unwraps are strictly forbidden as they will keep the UI waiting for a reply that never happens, usually stalling it. On top of that, panics show up in the `tokio` logs only which usually aren't provided if logs are handed in by hand, making these issues extra-hard to debug remotely.

In these situations, one can go with `fallible.map_or(true, |(num_ahead, _)| num_ahead == 0)`. *(Not actually sure what one should assume if this call fails)*